src/avif.c

```diff
@@ -189,7 +189,7 @@ void avifImageCopyNoAlloc(avifImage * dstImage, const avifImage * srcImage)
     dstImage->imir = srcImage->imir;
 }
 
-void avifImageCopySamples(avifImage * dstImage, const avifImage * srcImage, avifPlanesFlags planes)
+void avifImageCopyPlanes(avifImage * dstImage, const avifImage * srcImage, avifPlanesFlags planes)
```

Comment:
Current situation: There is an assertion, so `yuvFormat` must match. However in some rare cases it may not be the case; it would still be good to catch these issues in debug mode, but let them pass in production as they are not dangerous. That is why I consider both comments to be fine.

I agree the current situation described above could be improved. Feel free to either:
- Be stricter and always forbid mismatching `yuvRange`, either by forbidding HEIF/OBU mismatch or by using an `avifImage` view with the proper overloaded `yuvRange` to silence the assertion/error
- Modify the comments to explicitly allow `yuvRange` mismatch and remove the assertion
- Add a `allowYuvRangeMismatch` argument
- Leave things as is

Reply:
I haven't studied the issue fixed by https://github.com/AOMediaCodec/libavif/pull/1688, so I don't know which of your four suggestions is right. Anything we can do to eliminate the apparent contradiction between those two comments would be good.

Reply:
Done at https://github.com/AOMediaCodec/libavif/pull/1854.